sql/core/src/main/scala/org/apache/spark/sql/parquet/newParquet.scala

```diff
@@ -226,13 +247,14 @@ case class ParquetRelation2
         f -> new Footer(f.getPath, parquetMetadata)
       }.seq.toMap
 
-      partitionSpec = {
-        val partitionDirs = dataStatuses
+      partitionSpec = maybePartitionSpec.getOrElse {
+        val partitionDirs = leaves
```

Comment:
Is it possible that we will miss partitioning columns? If not, I think we can ignore entirely empty directories.